lib/eosknowledgeprivate/ekn-media-bin.c

```diff
@@ -0,0 +1,1462 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * ekn-media-bin.c
+ *
+ * Copyright (C) 2016 Endless Mobile, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Author: Juan Pablo Ugarte <ugarte@endlessm.com>
+ *
+ */
+
+#include "ekn-media-bin.h"
+#include <gst/gst.h>
+#include <epoxy/gl.h>
+#include <gdk/gdkkeysyms.h>
+
+#define USE_TMP_IMAGE            TRUE /* Packs an image before going fullscreen */
+#define PROP_AUTOHIDE_DEFAULT    2    /* Controls autohide timeout in seconds */
+
+#define EMB_ICON_SIZE            GTK_ICON_SIZE_BUTTON
+
+#define EMB_ICON_NAME_PLAY       "gtk-media-play"
+#define EMB_ICON_NAME_PAUSE      "gtk-media-pause"
+#define EMB_ICON_NAME_FULLSCREEN "view-fullscreen"
+#define EMB_ICON_NAME_RESTORE    "view-restore"
+
+struct _EknMediaBin
```

Comment:
Right, but this is not public since its defined in the .c file
Of course people could copy paste the struct in their code and access it but the same thing can be done with the private struct